packages/transformer/src/IModelTransformer.ts

```diff
@@ -1944,6 +1941,31 @@ export class IModelTransformer extends IModelExportHandler {
       ? targetElementId
       : undefined;
 
+    if (this._options.preserveElementIdsForFiltering) {
+      const isInvalid = !Id64.isValid(targetElementId);
+      const sourceInTargetElemProps = this.targetDb.elements.tryGetElementProps(
```

Comment:
Theres an opportunity for a slight optimization here, in that we dont need to call tryGetElementProps unless the first if statement is false.